examples/erc1155/lib.rs

```diff
@@ -0,0 +1,687 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(test, allow(dead_code))]
+#![cfg_attr(test, allow(unused_imports))]
+
+use ink_env::AccountId;
+use ink_lang as ink;
+use ink_prelude::vec::Vec;
+
+// This is the "magic" return value that we expect if a smart contract supports receiving ERC-1155
+// tokens.
+//
+// It is calculated with
+// `bytes4(keccak256("onERC1155Received(address,address,uint256,uint256,bytes)"))`, and corresponds
+// to 0xf23a6e61.
+const MAGIC_VALUE: [u8; 4] = [242, 58, 110, 97];
```

Comment:
I would maybe just call it "selector" then? xD
E.g. `ON_ERC_1155_RECEIVED_SELECTOR` or something like that.
It isn't a magic value since you even explained its value. :D

Reply:
In the ERC-1155 spec they call it "magic value", but yeah maybe it should have a proper name 😅